spring-boot/admin/src/main/java/eu/coatrack/admin/controllers/AdminController.java

```diff
@@ -81,570 +85,571 @@
+
+				if (colorForStatusCode == null) {
+					// there is no fixed color defined for this status code, set it based on the
+					// range
+					if (statusCode >= 200 && statusCode < 300) {
+						// lighter green
+						colorForStatusCode = new Color(0, 204, 0);
+					} else if (statusCode >= 300 && statusCode < 400) {
+						colorForStatusCode = Color.LIGHT_BLUE;
+					} else if (statusCode >= 404 && statusCode < 500) {
+						colorForStatusCode = Color.DARK_ORANGE;
+					} else if (statusCode >= 500 && statusCode < 600) {
+						// red
+						colorForStatusCode = new Color(255, 51, 51);
+					} else {
+						colorForStatusCode = Color.LIGHT_GRAY;
+					}
+				}
+				chartColors.add(colorForStatusCode);
+			}
+
+			DoughnutDataset dataset = new DoughnutDataset().setLabel("HTTP response codes")
+					.addBackgroundColors(chartColors.stream().toArray(Color[]::new)).setBorderWidth(2);
+			statisticsPerHttpStatusCodeList.forEach(stats -> dataset.addData(stats.getNoOfCalls()));
+			DoughnutData data = new DoughnutData().addDataset(dataset);
+			statisticsPerHttpStatusCodeList.forEach(stats -> data.addLabel(stats.getStatusCode().toString()));
+			return new DoughnutChart(data);
+		} else {
+			return new DoughnutChart();
+		}
+	}
+
+	@RequestMapping(value = "/dashboard/statsPerDayLineChart", method = GET, produces = "application/json")
+	@ResponseBody
+	public LineChart generateStatsPerDayLineChart(
+			@RequestParam("dateFrom") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate selectedTimePeriodStart,
+			@RequestParam("dateUntil") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate selectedTimePeriodEnd) {
+
+		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+		List<StatisticsPerDay> statsList = metricsAggregationCustomRepository
+				.getNoOfCallsPerDayForDateRange(selectedTimePeriodStart, selectedTimePeriodEnd, auth.getName());
+
+		// create a map with entries for all days in the given date range
+		Map<LocalDate, Long> callsPerDay = new TreeMap<>();
+		long timePeriodDurationInDays = ChronoUnit.DAYS.between(selectedTimePeriodStart, selectedTimePeriodEnd);
+		for (int i = 0; i <= timePeriodDurationInDays; i++) {
+			// put "0" as default, in case no calls are registered in database
+			callsPerDay.put(selectedTimePeriodStart.plusDays(i), 0l);
+		}
+
+		// add numbers from database, if any
+		statsList.forEach(statisticsPerDay -> {
+			callsPerDay.put(statisticsPerDay.getLocalDate(), statisticsPerDay.getNoOfCalls());
+		});
+
+		// create actual chart
+		LineDataset dataset = new LineDataset().setLabel("Total number of API calls per day")
+				.setBackgroundColor(Color.LIGHT_YELLOW).setBorderWidth(3);
+		LineData data = new LineData().addDataset(dataset);
+
+		callsPerDay.forEach((date, noOfCalls) -> {
+			data.addLabel(DateTimeFormatter.ISO_LOCAL_DATE.format(date));
+			dataset.addData(noOfCalls).addPointStyle(PointStyle.CIRCLE).addPointBorderWidth(2).setLineTension(0f)
```

Comment:
Here, it seems that you IDE is indenting automatically thus reducing the readability.
For better readability in the future, if there is  a stream, then it is better to put each step in a new line

Reply:
I didnt change these lines so, if there is a change could be just formating